.github/workflows/build-push.yml

```diff
@@ -0,0 +1,110 @@
+
+name: Build and Push Docker Image WEAC Codebase
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'python-flask/**'
+      - 'c-sharp-razor/**'
+      - 'rust-actix/**'
+      - 'ruby-sinatra/**'
+      - 'javascript-express/**'
+      - 'go-gin/**'
+      - 'go-gorilla/**'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        service:
+          - python-flask
+          - c-sharp-razor
+          - rust-actix
+          - ruby-sinatra
+          - javascript-express
+          - go-gin
+          - go-gorilla
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: '1'
```

Comment:
I think i would use fetch-depth 0 here to get the full history of the branch. Otherwise we only fetch most recent commit which is not ideal.

Reply:
true. Nice catch. But just be careful @simoonH when changing it because probably the commit sha that you are getting at the bottom of the file miight need to be changed as well

Reply:
Why would we like to have the full commit history? From what I know `fetch-depth=1` is the most sufficient for most cd pipelines. We minimize the amount of data the CI/CD runner has to process and store specially when we have a long history. Setting it to 0 would influence the performance. What was the reason you want to set it to 0?